src/proof/nova.rs

```diff
@@ -3631,4 +3631,77 @@ pub mod tests {
         );
         test_aux(s, &expr4, None, None, Some(error), None, 1, Some(lang));
     }
+
+    #[test]
+    fn test_issue424() {
+        let s = &mut Store::<Fr>::default();
+        let expected = s.num(5);
+        let terminal = s.get_cont_terminal();
+        test_aux::<Coproc<Fr>>(
+            s,
+            "(letrec ((char2int (lambda (digit)
+                           (if (eq #\\0 digit) 0 1)))
+         (str2int (letrec ((inner (lambda (acc xs)
+                                          (if (eq xs \"\") acc
+                                              (inner (+ (* 2 acc)
+                                                        (char2int (car xs)))
+                                                     (cdr xs))))))
+                          (inner 0))))
+        (str2int \"101\"))",
+            Some(expected),
+            None,
+            Some(terminal),
+            None,
+            147,
+            None,
+        );
+    }
+
+    #[test]
+    fn test_issue424_modified() {
+        let s = &mut Store::<Fr>::default();
+        let expected = s.num(5);
+        let terminal = s.get_cont_terminal();
+        test_aux::<Coproc<Fr>>(
+            s,
+            "(letrec ((char2int (lambda (digit)
+                           (if (eq #\\0 digit) 0 1)))
+         (inner (lambda (acc xs)
+                  (if (eq xs \"\") acc
+                    (inner (+ (* 2 acc)
+                              (char2int (car xs)))
+                           (cdr xs)))))
+         (str2int (inner 0)))
+        (str2int \"101\"))",
+            Some(expected),
+            None,
+            Some(terminal),
+            None,
+            147,
+            None,
+        );
+    }
+
+    #[test]
+    fn test_issue426() {
+        let s = &mut Store::<Fr>::default();
+        let nil = s.nil();
+        let strnil = s.str("");
+        let expected = s.cons(nil, strnil);
+        let terminal = s.get_cont_terminal();
+        test_aux::<Coproc<Fr>>(
+            s,
+            "(let ((f (lambda (x) nil))
+      (g (lambda (xs)
+                 (cons (f (car xs))
+                       (cdr xs)))))
+     (g \"0\"))",
```

Comment:
Can you please write example code so that it is correctly formatted? That either means adding a newline so code starts from the left margin, or else manually adding the amount of space required to appear correct post-rustfmt. I realize the latter is a little goofy, but at least the result is okay. Or, in this case, you could probably use the `lurk!` macro. However, that might require adjusting or expanding the test helpers.

Reply:
It's probably also worth reducing this to a more minimal example. You could include the original code in a comment, if you want — but I think you can demonstrate a more minimal failing test that exercises the problem (before the fix) and makes clearer what the issue is.

Reply:
Probably retaining the exact original source won't matter when a more minimal test case exists (in this instance).